### PR TITLE
raftstore: fix a stale read bug in follower read

### PR DESCRIPTION
**Pull request by hicqu**
Signed-off-by: qupeng <qupeng@pingcap.com>

###  What have you changed?
When a new leader has not committed to it's current term, it can not
response ReadIndex requests. In raft-rs the check is ok, but in
raftstore it's not. This PR adds the check in raftstore.

###  What is the type of the changes?
Bug fix.

###  How is the PR tested?
- [x] Integration test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?
No.

###  Does this PR affect `tidb-ansible`?
No.

###  Any examples? (optional)

### CODE REVIEW

File: src/raftstore/store/peer.rs [hunk 1]
```diff
@@ -826,7 +826,9 @@ impl Peer {
         // Here we hold up MsgReadIndex. If current peer has valid lease, then we could handle the
         // request directly, rather than send a heartbeat to check quorum.
         let msg_type = m.get_msg_type();
-        if msg_type == MessageType::MsgReadIndex {
+        let committed = self.raft_group.raft.raft_log.committed;
+        let expected_term = self.raft_group.raft.raft_log.term(committed).unwrap_or(0);
+        if msg_type == MessageType::MsgReadIndex && expected_term == self.raft_group.raft.term {
```
Comment (overvenus) [inline src/raftstore/store/peer.rs]: Please add some comments.

File: tests/integrations/raftstore/mod.rs [hunk 2]
```diff
@@ -6,6 +6,7 @@ mod test_compact_after_delete;
 mod test_compact_lock_cf;
 mod test_compact_log;
 mod test_conf_change;
+mod test_follower_read;
```
Comment (5kbpers) [inline tests/integrations/raftstore/mod.rs]: Prefer `test_replica_read`. Previous tests for replica read is placed in `storage/test_raftkv`, I think it's better to move them here.

Comment (hicqu) [reply tests/integrations/raftstore/mod.rs]: Let's move them in an individual PR.

**Pull request merged**